CMakeLists.txt

```diff
@@ -200,8 +255,10 @@ else()
   set(WITH_STDPAR_COMPILER false)
 endif()
 
-if(WITH_CUDA_BACKEND AND NOT WITH_CUDA_NVCXX_ONLY)
-  set(BUILD_CLANG_PLUGIN true)
+if(WITH_CUDA_BACKEND)
+  if(NOT WITH_CUDA_NVCXX_ONLY AND NOT ACPP_COMPILER_FEATURE_PROFILE STREQUAL "none")
+    set(BUILD_CLANG_PLUGIN true)
+  endif()
 endif()
 if(WITH_ROCM_BACKEND)
   set(BUILD_CLANG_PLUGIN true)
```

Comment:
If I follow the logic correctly, if the machine has ROCm installed, the `cmake .. -DACPP_COMPILER_FEATURE_PROFILE=none` will end up building the plugin. That does not look correct.

Same for `WITH_OPENCL_BACKEND` below.

Reply:
> If I follow the logic correctly, if the machine has ROCm installed, the cmake .. -DACPP_COMPILER_FEATURE_PROFILE=none will end up building the plugin. That does not look correct.

No, the plugin is built if `feature profile != none && !WITH_CUDA_NVCXX_ONLY`. So these two settings prevent the plugin to be built.

What does no longer exactly work as before is the `WITH_CUDA_NVCXX_ONLY` switch, since at the time where we initialize this in the custom-deprecated profile we don't know yet which backends need to be supported. But I think the interaction of this switch with SSCP and so on was poorly defined from the beginning.

For OpenCL/L0 you are right, but that is the same behavior as currently. The difficulty is to retain both the functionality of the old flags, as well as the new one.
We could enforce that when `feature profile == none` an error is emitted or so, but on the whole I think the full semantics of the new flag can really only shine once the old stuff can be removed.

Reply:
I've changed so that OpenCL and Level Zero don't pull in the clang plugin if the feature profile is "none" or "minimal".